docs/source/howto/data.rst

```diff
@@ -260,6 +260,359 @@ Finding data
+For that we need to make a query:
+
+.. code-block:: python
+
+    qb = QueryBuilder()
+    qb.append(Group, filters={'label':'promising_structures'}, tag='group')
+    qb.append(StructureData, with_group='group')
+
+The results of this query can be used, for instance, for subsequent simulations:
+
+.. code-block:: python
+
+    for structure in q.all(flat=True):
+        builder = SomeWorkChain.get_builder()
+        builder.structure = structure
+        ...
+        submit(builder)
+
+Note, however, that one can also use ``group.nodes`` to access the nodes of the group.
+To achieve the same result as above one would need to do something as follows:
+
+.. code-block:: python
+
+    group = load_group(label='promising_structures')
+    # Here make sure to include only structures, as group can contain any nodes.
+    structures = [s for s in group.nodes if isinstance(nodes, StructureData)]
+    for structure in structures:
+        builder = SomeWorkChain.get_builder()
+        builder.structure = structure
+        ...
+        submit(builder)
+
+
+For the postprocessing of the results, one can also make use of the groups.
+To find the structure that has a property ``property_a`` lower than ``1`` and at the same time belongs to the ``promising_structures`` group, one would need to build a query that looks as follows:
+
+.. code-block:: python
+
+    qb = QueryBuilder()
+    qb.append(Group, filters={'label':'promising_structures'}, tag='group')
+    qb.append(StructureData, with_group='group', tag='structure', project='*')
+    qb.append(SomeWorkChain, with_incoming='structure', tag='calculation')
+    qb.append(Dict, with_incoming='calculation', filters={'attributes.property_a':{'<':1}})
+
+As a result, ``qb.all(flat=True)`` would contain all the structures matching the criteria.
+
+Case III: use groups for data exporting
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Groups can be very handy for selecting the parts of the database for sharing.
+Suppose, you want to share all the simulations you've done for the structures in the ``promising_structures`` group.
+
+.. code-block:: console
+
+    $ verdi export create export.aiida -G promising_structures
+    STARTING EXPORT...
+    RETRIEVING NODES FROM GROUPS...
+    RETRIEVING LINKED NODES AND STORING LINKS...
+    STORING DATABASE ENTRIES...
+    Exporting a total of 1461 db entries, of which 988 nodes.
+    STORING NODE ATTRIBUTES AND EXTRAS...
+    STORING GROUP ELEMENTS...
+    STORING DATA...
+    STORING REPOSITORY FILES...
+    File written in       22.5 s.
+    Success: wrote the export archive file to export.aiida
+
+Keep in mind that such an export operation would not only export the structures that are part of the group, but also the nodes linked to them, following the rules  discussed in the :ref:`topics:provenance:consistency:traversal-rules` section.
```

Comment:
I don't think it's appropriate to address the reader like this, but I think it's very easy to make the language a bit less imposing:
```suggestion
Such an export operation would not only export the structures that are part of the group, but also the nodes linked to them, following the rules  discussed in the :ref:`topics:provenance:consistency:traversal-rules` section.
```